apps/server/src/shared/infra/identity-management/keycloak/service/keycloak-configuration.service.spec.ts

```diff
@@ -126,72 +127,158 @@ describe('configureIdentityProviders', () => {
 		kcApiClientIdentityProvidersMock.update.mockResolvedValue();
 		kcApiClientIdentityProvidersMock.del.mockResolvedValue();
 	});
 
-	beforeEach(() => {
-		repo.findAll.mockClear();
+	afterEach(() => {
+		systemService.findOidc.mockClear();
 		kcApiClientIdentityProvidersMock.find.mockClear();
 		kcApiClientIdentityProvidersMock.create.mockClear();
 		kcApiClientIdentityProvidersMock.update.mockClear();
 		kcApiClientIdentityProvidersMock.del.mockClear();
+		kcApiClientIdentityProvidersMock.updateMapper.mockClear();
+		kcApiClientIdentityProvidersMock.createMapper.mockClear();
 		configService.get.mockClear();
 	});
 
-	afterAll(() => {
-		repo.findAll.mockRestore();
-		kcApiClientIdentityProvidersMock.find.mockRestore();
-		kcApiClientIdentityProvidersMock.create.mockRestore();
-		kcApiClientIdentityProvidersMock.update.mockRestore();
-		kcApiClientIdentityProvidersMock.del.mockRestore();
-		configService.get.mockRestore();
-	});
+	describe('configureIdentityProviders', () => {
+		it('should read configs from database successfully', async () => {
+			const result = await service.configureIdentityProviders();
+			expect(result).toBeGreaterThan(0);
+			expect(systemService.findOidc).toBeCalled();
+		});
 
-	it('should read configs from database successfully', async () => {
-		const result = await service.configureIdentityProviders();
-		expect(result).toBeGreaterThan(0);
-		expect(repo.findAll).toBeCalled();
-	});
+		it('should create a configuration in Keycloak', async () => {
+			kcApiClientIdentityProvidersMock.find.mockResolvedValue([]);
+
+			const result = await service.configureIdentityProviders();
+			expect(result).toBe(1);
+			expect(kcApiClientIdentityProvidersMock.create).toBeCalledTimes(1);
 
-	it('should create a configuration in Keycloak', async () => {
-		kcApiClientIdentityProvidersMock.find.mockResolvedValue([]);
+			kcApiClientIdentityProvidersMock.find.mockResolvedValue(idps);
+		});
+		it('should update a configuration in Keycloak', async () => {
+			const result = await service.configureIdentityProviders();
+			expect(result).toBe(1);
+			expect(kcApiClientIdentityProvidersMock.update).toBeCalledTimes(1);
+		});
+		it('should delete a new configuration in Keycloak', async () => {
+			systemService.findOidc.mockResolvedValue([]);
 
-		const result = await service.configureIdentityProviders();
-		expect(result).toBe(1);
-		expect(kcApiClientIdentityProvidersMock.create).toBeCalledTimes(1);
+			const result = await service.configureIdentityProviders();
+			expect(result).toBe(1);
+			expect(kcApiClientIdentityProvidersMock.del).toBeCalledTimes(1);
 
-		kcApiClientIdentityProvidersMock.find.mockResolvedValue(idps);
-	});
-	it('should update a configuration in Keycloak', async () => {
-		const result = await service.configureIdentityProviders();
-		expect(result).toBe(1);
-		expect(kcApiClientIdentityProvidersMock.update).toBeCalledTimes(1);
+			systemService.findOidc.mockRestore();
```

Comment:
This resets the mock to the original implementation without a mock. I guess you wanted to use `mockReset` or `mockClear`

Reply:
Done